src/bin/cli.js

```diff
@@ -6,138 +6,155 @@ const chalk = require('chalk');
+};
+
+const inputLoopAsyncApi = (rd,asyncFile,regex) => {
+  const res = checkFilepath(asyncFile,regex);
+
+  if (!res) {
+    return new Promise((resolve) => {
+      rd.question('\nPlease provide an existent yaml or json file.It should abide by the asyncApi Spec.\nAsyncApi filepath:',(answer) => {
+        const inputLoop = (filepath) => {
+          const res = checkFilepath(filepath,regex);
+          if (!res) {
+            rd.question('Please fix errors and provide a correctly formatted and accessible file in filepath.\nAsyncApi Filepath:',(answer) => {
+              inputLoop(answer);
+            });
+          } else resolve(filepath);
+        };
+
+        inputLoop(answer);
+      });
+    });
+  }
+  return new Promise((resolve) => {
+    resolve(asyncFile);
   });
 };
 
 /**
- * Verifies the command line arguments, re-prompts in case of error. Parses file and returns the object representation.
+ * Verifies the command line arguments, re-prompts in case of error. Parses asyncApiF and returns the object representation.
  * @returns {Promise<*|null>}
  * @param rd
- * @param file
+ * @param asyncApiF
  * @param scenarioFile
  */
-const verifyInput_ParseAndLinkFiles =  async (rd, file,scenarioFile) => {
-  const handlingContext = this;
-  handlingContext.ready = false;
-  handlingContext.rd = rd;
-  handlingContext.file = file;
-  handlingContext.scenarioFile= scenarioFile;
+const verifyInput_getData =  async (rd, asyncApiF,scenarioFile) => {
   const yamlJsonRegex = RegExp(/^.*\.(json|yaml)$/, 'gm');
-  handlingContext.yamlJsonRegex = yamlJsonRegex;
 
   console.log(chalk.blueBright(`
   Async api Fluffy-robot
   `));
   console.log('\nWelcome ');
 
-  if (!!file) {
-    if (!String(file).match(yamlJsonRegex)) {
-      console.log('\nPlease provide a correctly formatted filepath ex: ./myAsyncApi.json ./myAsyncApi.yaml:');
-      inputLoopAsyncApi(handlingContext);
-    } else {
-      handlingContext.file = file;
-      handlingContext.ready = true;
-      if (!scenarioFile) {
-        inputLoopScenario(handlingContext);
-      } else if (!String(scenarioFile).match(yamlJsonRegex)) {
-        inputLoopScenario(handlingContext);
-      } else {
-        handlingContext.scenarioFile = scenarioFile;
-        handlingContext.scenarioReady = true;
-        parseAsyncApi(handlingContext);
-      }
-    }
-  } else {
-    console.log('\nFilepath not provided');
-    inputLoopAsyncApi(handlingContext);
-  }
+  asyncApiF = await inputLoopAsyncApi(rd,asyncApiF,yamlJsonRegex);
 
-  return handlingContext.ready ? await handlingContext.ParsedAndFormated : null;
+  scenarioFile = await inputLoopScenario(rd,scenarioFile,yamlJsonRegex);
+
+  const structuredData = await  scenarioParserAndConnector(asyncApiF,scenarioFile);
+
+  const availableServers = Object.keys(structuredData.servers);
+
+  structuredData.targetedServer = await inputLoopServer(availableServers);
+
+  return   structuredData;
```

Comment:
Extra space, are we missing some ESLint rules? 🤔

Reply:
i dont think so i have eslint file as in parser.js